### PR TITLE
Exposing attributes to make accessible menus and listboxes

### PR DESCRIPTION
**Pull request by piotrowskiLuc**
Added trait 'Option' in order to have role=option for listboxes.
Exposing attributes:
id, aria-labelledBy ,aria-haspopup, aria-controls to make accessible menus.

### CODE REVIEW

File: src/common/Types.ts [hunk 1]
```diff
@@ -463,6 +469,10 @@ export interface ButtonProps extends CommonStyledProps<ButtonStyleRuleSet>, Comm
     disableTouchOpacityAnimation?: boolean;
     activeOpacity?: number;
     underlayColor?: string;
+
+    // Web only.
+    id?: string; // Needed for accessibility to be able to use labelledBy attribute.
```
Comment (erictraut) [inline src/common/Types.ts]: Please correct spelling. Should be "labeledBy".

File: src/common/Types.ts [hunk 2]
```diff
@@ -562,6 +574,8 @@ export interface ViewPropsShared extends CommonProps, CommonAccessibilityProps {
     limitFocusWithin?: boolean; // Web-only, make the view and all focusable subelements not focusable when isFocusLimited state is true
 
     importantForLayout?: boolean; // Web-only, additional invisible DOM elements will be added to track the size changes faster
+    id?: string; // Web-only. Needed for accessibility.
+    ariaLabelledby?: string; // Web-only. Needed for accessibility.
```
Comment (erictraut) [inline src/common/Types.ts]: Please correct spelling. Should be "ariaLabeledBy".

File: src/web/AccessibilityUtil.ts [hunk 3]
```diff
@@ -95,6 +95,17 @@ export class AccessibilityUtil extends CommonAccessibiltiyUtil {
         // as we dont want to pollute the dom with "aria-checked = false" for every falsy condition
         return undefined;
     }
+
+    accessibilityTraitToAriaHasPopup(traits: Types.AccessibilityTrait | Types.AccessibilityTrait[]|undefined) {
```
Comment (erictraut) [inline src/web/AccessibilityUtil.ts]: Nit: For consistency, add spaces around "|" operator.

File: src/common/Types.ts [hunk 4]
```diff
@@ -547,6 +547,8 @@ export interface TextPropsShared extends CommonProps {
     elevation?: number;
 
     onPress?: (e: SyntheticEvent) => void;
+
+    id?: string; // Web only. Needed for accessibility.
```
Comment (erictraut) [inline src/common/Types.ts]: Please add to appropriate documentation (".md") file. Props are listed in alphabetical order in the documentation. Docs also explain the default value. I think the default value in this case is undefined.

File: src/common/Types.ts (hunk 1 above)
Comment (piotrowskiLuc) [reply src/common/Types.ts]: I've addressed the comments and added documentation to respective *.md files

Comment (mshoho) [reply src/common/Types.ts]: In the standard it's double «l» — https://www.w3.org/TR/wai-aria/states_and_properties#aria-labelledby

Comment (piotrowskiLuc) [reply src/common/Types.ts]: Lets wait with merging it. I think I still need to expose aria-expanded attribute

Comment (piotrowskiLuc) [reply src/common/Types.ts]: Actually lets go without aria-expanded for now. We can add it later if we see it's needed.

**Pull request merged**